robotoff/prediction/ocr/packager_code.py

```diff
@@ -34,75 +34,136 @@ def process_fsc_match(match) -> str:
+
+    for (USDA_code_keyword, _) in processor.extract_keywords(text):
+        USDA_code = USDA_code_keyword
+    return USDA_code
+
+
+USDA_CODE_KEYWORD_PROCESSOR_STORE = CachedStore(
+    fetch_func=generate_USDA_code_keyword_processor, expiration_interval=None
+)
+
+
+PACKAGER_CODE = {
+    "fr_emb": [
+        OCRRegex(
+            re.compile(r"emb ?(\d ?\d ?\d ?\d ?\d) ?([a-z])?(?![a-z0-9])"),
+            field=OCRField.text_annotations,
+            lowercase=True,
+            processing_func=process_fr_emb_match,
+        ),
+    ],
+    "fsc": [
+        OCRRegex(
+            re.compile(r"fsc.? ?(c\d{6})"),
+            field=OCRField.text_annotations,
+            lowercase=True,
+            processing_func=process_fsc_match,
+        ),
+    ],
+    "eu_fr": [
+        OCRRegex(
+            re.compile(
+                r"fr (\d{2,3}|2[ab])[\-\s.](\d{3})[\-\s.](\d{3}) (ce|ec)(?![a-z0-9])"
+            ),
+            field=OCRField.full_text_contiguous,
+            lowercase=True,
+            processing_func=process_fr_packaging_match,
         ),
-        field=OCRField.full_text_contiguous,
-        lowercase=True,
-        processing_func=process_fr_packaging_match,
-    ),
-    "eu_de": OCRRegex(
-        re.compile(
-            r"de (bb|be|bw|by|hb|he|hh|mv|ni|nw|rp|sh|sl|sn|st|th)[\-\s.](\d{1,5})[\-\s.] ?(eg|ec)(?![a-z0-9])"
+    ],
+    "eu_de": [
+        OCRRegex(
+            re.compile(
+                r"de (bb|be|bw|by|hb|he|hh|mv|ni|nw|rp|sh|sl|sn|st|th)[\-\s.](\d{1,5})[\-\s.] ?(eg|ec)(?![a-z0-9])"
+            ),
+            field=OCRField.full_text_contiguous,
+            lowercase=True,
+            processing_func=process_de_packaging_match,
         ),
-        field=OCRField.full_text_contiguous,
-        lowercase=True,
-        processing_func=process_de_packaging_match,
-    ),
-    "rspo": OCRRegex(
-        re.compile(r"(?<!\w)RSPO-\d{7}(?!\d)"),
-        field=OCRField.full_text_contiguous,
-        lowercase=False,
-    ),
-    "fr_gluten": OCRRegex(
-        re.compile(r"FR-\d{3}-\d{3}"),
-        field=OCRField.full_text_contiguous,
-        lowercase=False,
-    ),
+    ],
+    "rspo": [
+        OCRRegex(
+            re.compile(r"(?<!\w)RSPO-\d{7}(?!\d)"),
+            field=OCRField.full_text_contiguous,
+            lowercase=False,
+        ),
+    ],
+    "fr_gluten": [
+        OCRRegex(
+            re.compile(r"FR-\d{3}-\d{3}"),
+            field=OCRField.full_text_contiguous,
+            lowercase=False,
+        ),
+    ],
+    "USDA": [
+        # To match the USDA like "EST. 522"
+        OCRRegex(
+            re.compile(r"EST\.*\s*\d{1,5}[A-Z]{0,3}\.*"),
+            field=OCRField.full_text_contiguous,
+            lowercase=False,
+            processing_func=process_USDA_match_to_flashtext,
+        ),
+        # To match the USDA like "V34626" or "M34614 + P34614 + V34614"
+        OCRRegex(
+            re.compile(r"[A-Z]\d{1,5}[A-Z]?(\s*\+\s*[A-Z]\d{1,5}[A-Z]?){0,3}"),
```

Comment:
Do we really need to capture the " + " in one go, and not just each code ? This would simplify the task I think.

Reply:
Actually, I did this for the same reason I did what you saw as a mistake two comments upwards. I thought that some products had codes which were explicitly "M34614 + P34614 + V34614". That's what written in the data @teolemon put in the [Add support for the USDA packager codes #3704](https://github.com/openfoodfacts/openfoodfacts-server/issues/3704), [Available as an XLSX](https://docs.google.com/spreadsheets/d/1W7Z7R_wc5Rj9AabCr5hk5-JHgi-cJCxO/edit#gid=381973512).